src/google/protobuf/compiler/csharp/csharp_field_base.cc

```diff
@@ -57,24 +57,41 @@ void FieldGeneratorBase::SetCommonFieldVariables(
   // repeated fields varies by wire format. The wire format is encoded in the bottom 3 bits, which
   // never effects the tag size.
   int tag_size = internal::WireFormat::TagSize(descriptor_->number(), descriptor_->type());
+  int part_tag_size = tag_size;
   if (descriptor_->type() == FieldDescriptor::TYPE_GROUP) {
-    tag_size /= 2;
+    part_tag_size /= 2;
   }
   uint tag = internal::WireFormat::MakeTag(descriptor_);
   uint8 tag_array[5];
   io::CodedOutputStream::WriteTagToArray(tag, tag_array);
   string tag_bytes = SimpleItoa(tag_array[0]);
-  for (int i = 1; i < tag_size; i++) {
+  for (int i = 1; i < part_tag_size; i++) {
     tag_bytes += ", " + SimpleItoa(tag_array[i]);
   }
 
-  (*variables)["access_level"] = "public";
   (*variables)["tag"] = SimpleItoa(tag);
   (*variables)["tag_size"] = SimpleItoa(tag_size);
   (*variables)["tag_bytes"] = tag_bytes;
 
+  if (descriptor_->type() == FieldDescriptor::Type::TYPE_GROUP) {
```

Comment:
This change is related to group support not extension?

Reply:
Same as above, it's cleanup that I found necessary to implement the changes for extensions.